demo/android/OrcaDemo/orca-demo-app/src/main/java/ai/picovoice/orcademo/MainActivity.java

```diff
@@ -322,25 +350,199 @@ private void stopPlayback() {
+        try {
+            runOnUiThread(() -> {
+                setUIState(UIState.EDIT);
+                infoTextView.setText("");
+                streamTextView.setVisibility(View.INVISIBLE);
+                synthesizeEditText.setVisibility(View.VISIBLE);
+            });
+        } catch (Exception e) {
+            displayError(e.toString());
         }
     }
 
+    private void runStreamSynthesis(final String text) {
+        runOnUiThread(() -> {
+            setUIState(UIState.PLAYBACK);
+            infoTextView.setText("Streaming...");
+            streamTextView.setVisibility(View.VISIBLE);
+            streamSecsTextView.setVisibility(View.VISIBLE);
+            synthesizeEditText.setVisibility(View.INVISIBLE);
+        });
+
+        AtomicBoolean isStreamingText = new AtomicBoolean(false);
+        ArrayList<String> textStream = new ArrayList<>();
+
+        AtomicBoolean isQueueingStreamingPcm = new AtomicBoolean(false);
+        ConcurrentLinkedQueue<short[]> pcmQueue = new ConcurrentLinkedQueue<>();
+        CountDownLatch latch = new CountDownLatch(1);
+
+        executor.submit(() -> {
+            isStreamingText.set(true);
+
+            String[] words = text.split(" ");
+            for (String word : words) {
+                word += " ";
+                try {
+                    String finalWord = word;
+                    mainHandler.post(() -> {
+                        textStream.add(finalWord);
+                        streamTextView.append(finalWord);
+                    });
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            isStreamingText.set(false);
+        });
+
+        executorStreamingSynthesis.submit(() -> {
+            try {
+                mainHandler.post(() -> {
+                    streamTextView.setText("");
+                    streamSecsTextView.setText("Seconds of audio synthesized: 0.000s");
+                    synthesizeButton.setEnabled(false);
+                });
+
+                int numIterations = 0;
+                boolean isPcmPlayStarted = false;
+                float secs = 0;
+                isQueueingStreamingPcm.set(true);
+
+                while (isStreamingText.get() || !textStream.isEmpty()) {
+                    if (!textStream.isEmpty()) {
+                        String word = textStream.remove(0);
+                        try {
+                            short[] pcm = orcaStream.synthesize(word);
+                            if (pcm != null && pcm.length > 0) {
+                                pcmQueue.add(pcm);
+                                secs += (float) pcm.length / orca.getSampleRate();
+                                float finalSecs = secs;
+                                mainHandler.post(() -> streamSecsTextView.setText("Seconds of audio synthesized: " + String.format("%.3f", finalSecs) + "s"));
```

Comment:
You can string format the whole string 
```suggestion
                                mainHandler.post(() -> streamSecsTextView.setText(String.format("Seconds of audio synthesized: %.3fs", finalSecs));
```